azure-keyvault/azure/keyvault/custom/key_vault_client.py

```diff
@@ -6,12 +6,43 @@
 import uuid
 from msrest.pipeline import ClientRawResponse
 
+from .key_vault_authentication import KeyVaultAuthBase, KeyVaultAuthentication
 from ..key_vault_client import KeyVaultClient as KeyVaultClientBase
 from ..models import KeyVaultErrorException
+from msrestazure.azure_active_directory import AADTokenCredentials
 
 
 class CustomKeyVaultClient(KeyVaultClientBase):
 
+    def __init__(self, credentials):
+        """The key vault client performs cryptographic key operations and vault operations against the Key Vault service.
+
+        :ivar config: Configuration for client.
+        :vartype config: KeyVaultClientConfiguration
+
+        :param credentials: Credentials needed for the client to connect to Azure.
+        :type credentials: :mod:`A msrestazure Credentials
+         object<msrestazure.azure_active_directory>` or :mod:`A KeyVaultAuthentication
+         object<key_vault_authentication>` 
+        """
+        self._inner_creds = None
+
+        # if the supplied credentials instance is not derived from KeyVaultAuthBase but is an AADTokenCredentials instance
+        if not isinstance(credentials, KeyVaultAuthBase) and isinstance(credentials, AADTokenCredentials):
```

Comment:
`AADMixin`, not `AADTokenCredentials`. This one is deprecated and not in the hierarchy of `ServicePrincipalCredentials`